### ISSUE TITLE
feat: file-log-plugin should combine with log-rotate

### ISSUE THREAD

**Issue by liangliang4ward** [apache/apisix#6709]
### Description

I found that file-log-plugin just write one file. I think we can combine with log-rotate plugin to split file

**Comment by liangliang4ward** [comment]
I mean ,the file-log-plugin write only one file. I think the file will be very large.  we should split them

**Comment by tzssangglass** [comment]
good idea,  we should make the file logger itself support log file rotation.. cc @spacewander

**Comment by spacewander** [comment]
This is not easy to combine these two plugins. But we can make a try.

**Comment by tzssangglass** [comment]
> This is not easy to combine these two plugins. But we can make a try.

Maybe we can make file-logger support log file rotation (without modifying the log-rotate plugin, or extracting some common functions)?

**Comment by liangliang4ward** [comment]
> > This is not easy to combine these two plugins. But we can make a try.
> 
> Maybe we can make file-logger support log file rotation (without modifying the log-rotate plugin, or extracting some common functions)?

yes, I think it can be.

**Comment by AlphaChen** [comment]
i meet same problem;

**Comment by tokers** [comment]
What if we support configuring the file names for both the log-rotate plugin and file-logger plugin?

**Comment by AlphaChen** [comment]
> What if we support configuring the file names for both the log-rotate plugin and file-logger plugin?

yes we need logrotate

**Comment by spacewander** [comment]
I think of this a bit.
What about configuring the log rotate configuration in file-logger's metadata? And let the plugin level conf refers the conf in the metadata. Therefore we can:
1. make the file log rotate conf can be changed dynamically
2. don't need to configure the log rotate repeatedly in each plugin level conf

CC @monkeyDluffy6017

**Comment by monkeyDluffy6017** [comment]
I just thought about your proposal, so we could get metadata from timer and roate file log in timer ?

**Comment by monkeyDluffy6017** [comment]
I just thought about your proposal, so we could get metadata from timer and roate file log in timer ? @spacewander

**Comment by tokers** [comment]
> > What if we support configuring the file names for both the log-rotate plugin and file-logger plugin?
> 
> yes we need logrotate

I mean, we should avoid doing duplicated things, if we can combine the two plugins and let them to work for the needs, why do we need to modify the file-logger plugin to support a supported feature?

**Comment by monkeyDluffy6017** [comment]
@tokers maybe it's not good idea to link the 2 plugins, it's better to make them independent ?

**Comment by tokers** [comment]
> @tokers maybe it's not good idea to link the 2 plugins, it's better to make them independent ?

The combine means using them together.

**Comment by spacewander** [comment]
Putting the configuration into log-rotate is possible, but there are some cons so I personally don't like this idea.

Here is the limitation:
1. the feature is meaningful only when used with other plugins like file-logger.
2. it requires rewriting the current implementation. The current implementation uses a static conf file to configure the log rotate options and hardcodes the log file's name. The workload of modification isn't lower than modifying the file-logger.

**Comment by cataglyphis** [comment]
Any update to this issue? We find that apisix file-logger save all the logs to same file which is too large, make the k8s node unavailable.
